#### 1. 
Как относятся друг к другу множество и последовательность? (в ответе использовать слова типа: часть, целое, общее, частное,
родитель, дочерний субъект и т.д.)
#### ОТВЕТ:
Последовательность – выборка, созданная на упорядочненном множестве натуральных чисел, т.е. множество натуральных чисел (выступает в роли родителя) отображается на последовательность по определённому закону (порождает последовательность как дочерний субъект).

#### 2. 
Прочитать высказывания математической логики, построить их отрицания и *установить истинность.
#### ОТВЕТ

$1.$ Для любого y из отрезка [0; 1] значение функции sgn(y) равно 1.   

Отрицание:

Существуют такие y из отрезка [0; 1], что значение функции sgn(y) не равно 1.

$2.$ Для любого n, принадлежащего множеству натуральных чисел, больших 2, существуют такие x, y, z из множества натуральных чисел, что x в степени n равен сумме y в степени n и z в степени n.

Отрицание: 

Существует такое n, принадлежащее множеству натуральных чисел, больших 2, для которого справедливо, что для любых x, y, z (из множества натуральных чисел) x в степени n не равен сумме y в степени n и z в степени n.

$3.$ Для любого вещественного числа x существует такое множество X, принадлежащее множеству вещественных чисел, что все элементы множества X больше x.

Отрицание: 

Существует такое вещественное число x для которого любое множество X, принадлежащее множеству вещественных чисел, будет таким, что все элементы множества X будут меньше x.

$4.$ Для любого комплексного числа x не существует комплексного числа y, такого чтобы: x > y  или y > x.

Отрицание: 

Существует такое комплексное число x для которого существует комплексное число y, такое, что справедливо: $x \geq y$  и $y \geq x$.

$5.$ Для любого y из отрезка $[0; \pi/2]$ существует такое $\varepsilon > 0$, что $sin(y) < sin(y + \varepsilon)$

Отрицание: 

Существует такой y из отрезка $[0; \pi/2]$, что для любого $\varepsilon \leq 0$ будет выполняться $sin(y) \geq sin(y + \varepsilon)$

$6.$ Для любого y из отрезка $[0; \pi)$ существует такое $\varepsilon > 0$, что $cos(y) > cos(y + \varepsilon)$

Отрицание: 

Существует такой y из отрезка $[0; \pi)$, что для любого $\varepsilon \leq 0$ будет справедливо $cos(y) \leq cos(y + \varepsilon)$

$7.$ Существует такой x, который не входит в множество, состоящее из натуральных, целых, рациональных, вещественных и комплексных чисел.

Отрицание: 

Любой x входит в множество, состоящее из натуральных, целых, рациональных, вещественных и комплексных чисел.

#### 3.  &  4.
Даны три множества a,b и с. Необходимо выполнить все изученные виды бинарных операций над всеми комбинациями множеств.

#### ОТВЕТ:


Под всеми комбинациями множеств будем считать все возможные размещения из n по k: $A_3^1 + A_3^2 + A_3^3$

In [1]:
import numpy as np

def arrangement(n, k):
    return np.math.factorial(n)/np.math.factorial(n - k)

A = arrangement(3, 1) + arrangement(3, 2) + arrangement(3, 3)
A

15.0

Всего 15 возможных размещений.
Перечислим.

In [2]:
from itertools import permutations

P = []

for j in range(1,4):
    for item in permutations('abc', j):
        P.append(''.join(item))
        
P

['a',
 'b',
 'c',
 'ab',
 'ac',
 'ba',
 'bc',
 'ca',
 'cb',
 'abc',
 'acb',
 'bac',
 'bca',
 'cab',
 'cba']

Изученые бинарные операции:    
пересечение, объединение, разность, симметрическая разность, декартово произведение

Бинарные комбинации возможны при участии не менее 2-х элементов, т.е. первые 3 члена из списка рассматриваться не будут.   
Зададим элементы множеств:

In [3]:
a = set('big')
b = set('band')
c = set('mafia')

Ограничение при использовании set - удаляются задваиваемые элементы внутри множества

Пересечение ------------------------

In [4]:
a&b

{'b'}

In [5]:
a&c

{'i'}

In [6]:
b&c

{'a'}

In [7]:
a&b&c

set()

Объединение --------------------- 

In [8]:
a|b

{'a', 'b', 'd', 'g', 'i', 'n'}

In [9]:
a|c

{'a', 'b', 'f', 'g', 'i', 'm'}

In [10]:
b|c

{'a', 'b', 'd', 'f', 'i', 'm', 'n'}

In [11]:
a|b|c

{'a', 'b', 'd', 'f', 'g', 'i', 'm', 'n'}

Разность --------------------

In [12]:
a - b

{'g', 'i'}

In [13]:
b - a

{'a', 'd', 'n'}

In [14]:
a - c

{'b', 'g'}

In [15]:
c - a

{'a', 'f', 'm'}

In [16]:
b - c

{'b', 'd', 'n'}

In [17]:
c - b

{'f', 'i', 'm'}

In [18]:
a - b - c

{'g'}

In [19]:
a - c - b

{'g'}

In [20]:
b - a - c

{'d', 'n'}

In [21]:
b - c - a

{'d', 'n'}

In [22]:
c - a - b

{'f', 'm'}

In [23]:
c - b - a

{'f', 'm'}

Симметрическая разность -----------------------

In [24]:
a^b

{'a', 'd', 'g', 'i', 'n'}

In [25]:
a^c

{'a', 'b', 'f', 'g', 'm'}

In [26]:
b^c

{'b', 'd', 'f', 'i', 'm', 'n'}

In [27]:
a^b^c

{'d', 'f', 'g', 'm', 'n'}

Декартово произведение ------------------

In [28]:
lst = [list(a), list(b), list(c)]
lst

[['i', 'g', 'b'], ['d', 'n', 'b', 'a'], ['i', 'm', 'f', 'a']]

In [29]:
from itertools import product
k = 1
for i in product(*lst):
    print(f'N {k} - {i}')
    k += 1

N 1 - ('i', 'd', 'i')
N 2 - ('i', 'd', 'm')
N 3 - ('i', 'd', 'f')
N 4 - ('i', 'd', 'a')
N 5 - ('i', 'n', 'i')
N 6 - ('i', 'n', 'm')
N 7 - ('i', 'n', 'f')
N 8 - ('i', 'n', 'a')
N 9 - ('i', 'b', 'i')
N 10 - ('i', 'b', 'm')
N 11 - ('i', 'b', 'f')
N 12 - ('i', 'b', 'a')
N 13 - ('i', 'a', 'i')
N 14 - ('i', 'a', 'm')
N 15 - ('i', 'a', 'f')
N 16 - ('i', 'a', 'a')
N 17 - ('g', 'd', 'i')
N 18 - ('g', 'd', 'm')
N 19 - ('g', 'd', 'f')
N 20 - ('g', 'd', 'a')
N 21 - ('g', 'n', 'i')
N 22 - ('g', 'n', 'm')
N 23 - ('g', 'n', 'f')
N 24 - ('g', 'n', 'a')
N 25 - ('g', 'b', 'i')
N 26 - ('g', 'b', 'm')
N 27 - ('g', 'b', 'f')
N 28 - ('g', 'b', 'a')
N 29 - ('g', 'a', 'i')
N 30 - ('g', 'a', 'm')
N 31 - ('g', 'a', 'f')
N 32 - ('g', 'a', 'a')
N 33 - ('b', 'd', 'i')
N 34 - ('b', 'd', 'm')
N 35 - ('b', 'd', 'f')
N 36 - ('b', 'd', 'a')
N 37 - ('b', 'n', 'i')
N 38 - ('b', 'n', 'm')
N 39 - ('b', 'n', 'f')
N 40 - ('b', 'n', 'a')
N 41 - ('b', 'b', 'i')
N 42 - ('b', 'b', 'm')
N 43 - ('b', 'b', 'f')
N 44 - ('b', 'b', 'a

#### 4.
Даны 4 последовательности.   
Необходимо:   
1) исследовать их на монотонность;   
2) исследовать на ограниченность;   
3) найти пятый по счету член.

#### ОТВЕТ:

$1.$ ----------------------------

In [30]:
def a_n(n):
    return 2*n - 1

In [31]:
n = 1000

a1 = a_n(n) - a_n(n+1)
a2 = a_n(10*n) - a_n(10*n+1)
a3 = a_n(100*n) - a_n(100*n+1)
a1, a2, a3

(-2, -2, -2)

Последовательность монотонно возрастающая

In [32]:
n = 1
a_n(n)

1

Последовательность ограничена снизу числом 1

In [33]:
a_n(5)

9

Пятый член последовательности = 9

$2.$ ----------------------------

In [34]:
def b_n(n):
    return 1/(1 - n)

In [35]:
n = 1000

b1 = b_n(n) - b_n(n+1)
b2 = b_n(10*n) - b_n(10*n+1)
b3 = b_n(100*n) - b_n(100*n+1)
b1, b2, b3

(-1.0010010010009715e-06, -1.0001000100008477e-08, -1.0000100000901087e-10)

Последовательность монотонно возрастающая

In [36]:
n = 2
b_n(2)

-1.0

Последовательность ограничена снизу -1, сверху 0

In [37]:
b_n(6)

-0.2

Пятый член последовательности = -0.2

$3.$ ----------------------------

In [38]:
def c_n(n):
    return (-1)**n + np.sqrt(2*n)

In [39]:
n = 1000

c1 = c_n(n) - c_n(n+1)
c2 = c_n(10*n) - c_n(10*n+1)
c3 = c_n(100*n) - c_n(100*n+1)
c1, c2, c3

(1.9776449076016078, 1.992929108955991, 1.9977639376126604)

Последовательность монотонно убывающая

In [40]:
n = 1
c_n(n)

0.41421356237309515

Последовательность ограничена снизу числом 0.41421356237309515

In [41]:
c_n(5)

2.1622776601683795

Пятый член последовательности = 2.1622776601683795

$4.$ ----------------------------

In [42]:
def d_n(n):
    return (-1)**(2*n) + 1/(n**2)

In [43]:
n = 1000

d1 = d_n(n) - d_n(n+1)
d2 = d_n(10*n) - d_n(10*n+1)
d3 = d_n(100*n) - d_n(100*n+1)
d1, d2, d3

(1.9970038955818836e-09, 1.999733711954832e-12, 1.9984014443252818e-15)

Последовательность монотонно убывающая

In [44]:
n = 1
d_n(n)

2.0

Последовательность ограничена снизу числом 1, сверху числом 2

In [45]:
d_n(5)

1.04

Пятый член последовательности = 1.04

#### 5.
найти 12-й член заданной неявно последовательности $$a_1 = 128$$ $$a_{n+1} - a_n = 6$$
#### ОТВЕТ:

$a_2 = 6 + a_1$    
$a_3 = 6 + a_2 = 6 \cdot 2 + a_1$  
...  
$a_{12} = 6 \cdot (12 - 1) + a_1$

In [46]:
a_1 = 128
a_12 = 6*11 + 128
a_12

194